lectures/markov_chains.md

```diff
@@ -326,84 +312,105 @@ In these exercises, we'll take the state space to be $S = 0,\ldots, n-1$.
+1. At time $t=0$, draw a realization of $X_0$ from the distribution $\psi$.
 1. At each subsequent time $t$, draw a realization of the new state $X_{t+1}$ from $P(X_t, \cdot)$.
 
 (That is, draw from row $X_t$ of $P$.)
 
-To implement this simulation procedure, we need a method for generating draws from a discrete distribution.
+To implement this simulation procedure, we need a method for generating draws
+from a discrete distribution.
 
 For this task, we'll use `random.draw` from [QuantEcon](http://quantecon.org/quantecon-py).
 
 To use `random.draw`, we first need to convert the probability mass function
 to a cumulative distribution
 
-TODO -- link to distributions lecture
-
 ```{code-cell} ipython3
 ψ = (0.3, 0.7)           # probabilities over {0, 1}
 cdf = np.cumsum(ψ)       # convert into cumulative distribution
 qe.random.draw(cdf, 5)   # generate 5 independent draws from ψ
 ```
 
++++ {"user_expressions": []}
+
 We'll write our code as a function that accepts the following three arguments
 
 * A stochastic matrix `P`
-* An initial state `init`
+* An initial distribution `ψ`
 * A positive integer `sample_size` representing the length of the time series the function should return
 
 ```{code-cell} ipython3
-def mc_sample_path(P, ψ_0=None, sample_size=1_000):
+def mc_sample_path(P, ψ=None, sample_size=1_000):
 
     # set up
     P = np.asarray(P)
     X = np.empty(sample_size, dtype=int)
 
     # Convert each row of P into a cdf
     n = len(P)
-    P_dist = [np.cumsum(P[i, :]) for i in range(n)]
+    P_dist = np.cumsum(P, axis=1)  # Convert rows into cdfs
 
     # draw initial state, defaulting to 0
-    if ψ_0 is not None:
-        X_0 = qe.random.draw(np.cumsum(ψ_0))
+    if ψ is not None:
+        X_0 = qe.random.draw(np.cumsum(ψ))
     else:
         X_0 = 0
 
     # simulate
     X[0] = X_0
     for t in range(sample_size - 1):
-        X[t+1] = qe.random.draw(P_dist[X[t]])
+        X[t+1] = qe.random.draw(P_dist[X[t], :])
 
     return X
 ```
 
++++ {"user_expressions": []}
+
 Let's see how it works using the small matrix
 
 ```{code-cell} ipython3
 P = [[0.4, 0.6],
      [0.2, 0.8]]
 ```
 
-As we'll see later, for a long series drawn from `P`, the fraction of the sample that takes value 0 will be about 0.25.
++++ {"user_expressions": []}
+
+Here's a short time series.
+
+```{code-cell} ipython3
+mc_sample_path(P, ψ=[1.0, 0.0], sample_size=10)
+```
 
-Moreover, this is true, regardless of the initial distribution from which
++++ {"user_expressions": []}
+
+It can be shown that for a long series drawn from `P`, the fraction of the
+sample that takes value 0 will be about 0.25.
+
+(We will explain why {ref}`below <ergodicity>`.)
+
+Moreover, this is true regardless of the initial distribution from which
 $X_0$ is drawn.
 
 The following code illustrates this
 
 ```{code-cell} ipython3
-X = mc_sample_path(P, ψ_0=[0.1, 0.9], sample_size=100_000)
+X = mc_sample_path(P, ψ=[0.1, 0.9], sample_size=1_000_000)
```

Comment:
Hi @jstac,

Thanks for reminding me about consistencies in notations in a recent issue.

I just noticed that you were avoiding the use of `ψ_0` in this function. I have used this convention in other functions as well. Is there any particular reason why we are avoiding `ψ_0`? If there is, I can change my other functions to fit this convention.

Many thanks in advance.